app/controllers/dashboard/notifications_controller.rb

```diff
@@ -52,7 +52,12 @@ def new
         @notification = Notification.new
       end
 
-        @message = @notification.messages.new(to: params[:identity_id])
+      if Identity.find(params[:identity_id]) == current_user
```

Comment:
ActiveRecord delegates `==` to comparing the `:id` columns, so technically you could write `if params[:identity_id] == current_user.id` here instead, with the benefit of not hitting the database with `#find`. Also, in the event that `params[:identity_id]` refers to `current_user`, are you still wanting to run line 60?